.github/workflows/build.yml

```
@@ -0,0 +1,305 @@
+name: build
+
+env:
+  SGX_MODE: HW
+  IAS_MODE: PROD
+  RUST_BACKTRACE: full
+  CONSENSUS_ENCLAVE_CSS: /var/tmp/consensus-enclave.css
+  INGEST_ENCLAVE_CSS: /var/tmp/ingest-enclave.css
+
+on:
+  push:
+    tags:
+      - v*
```

I imagine we want to enable this for main and PRs, since the steps check for `startsWith(refs/tags/v)`.
```suggestion
on:
  pull_request:
  push:
    tags: v*
    branches:
      - master
      - main
      - develop
  workflow_dispatch:
```

---

so, we IMO we really only wanna build for things that get tagged, and simply test/lint otherwise

---

ah, I see now, so yeah this was a port of some work done by Marco on the full-service-mirror, so can probably just remove the if check